testing/e2e/stand_alone_test.go

```diff
@@ -120,3 +120,130 @@ func (suite *StandAloneSuite) TestWithSecretFiles() {
+
+	bCtx, bCancel := context.WithCancel(context.Background())
+	defer bCancel()
+
+	// Run the fleet-server binary, cancelling context should stop process
+	cmd := exec.CommandContext(bCtx, suite.binaryPath, "-c", filepath.Join(dir, "config.yml"))
+	cmd.Cancel = func() error {
+		return cmd.Process.Signal(syscall.SIGTERM)
+	}
+	cmd.Env = []string{"GOCOVERDIR=" + suite.coverPath}
+	err = cmd.Start()
+	suite.Require().NoError(err)
+
+	ctx, cancel := context.WithTimeout(bCtx, time.Minute)
+	suite.FleetServerStatusOK(ctx, "https://localhost:8220")
+	cancel()
+
+	enrollmentKey := suite.GetEnrollmentTokenForPolicyID(bCtx, "dummy-policy")
+
+	// Run subtests here
+	suite.Run("test status unauthenicated", func() {
+		ctx, cancel := context.WithCancel(bCtx)
+		defer cancel()
+		tester := &ClientAPITester{
+			suite.Suite,
+			ctx,
+			suite.client,
+			"https://localhost:8220",
+		}
+		tester.TestStatus("")
+	})
+
+	suite.Run("test status authenicated", func() {
+		ctx, cancel := context.WithCancel(bCtx)
+		defer cancel()
+		tester := &ClientAPITester{
+			suite.Suite,
+			ctx,
+			suite.client,
+			"https://localhost:8220",
+		}
+		tester.TestStatus(enrollmentKey)
+	})
+
+	suite.Run("test enroll checkin ack", func() {
+		ctx, cancel := context.WithCancel(bCtx)
+		defer cancel()
+		tester := &ClientAPITester{
+			suite.Suite,
+			ctx,
+			suite.client,
+			"https://localhost:8220",
+		}
+
+		suite.T().Log("test enrollment")
+		agentID, agentKey := tester.TestEnroll(enrollmentKey)
+		suite.VerifyAgentInKibana(ctx, agentID)
+
+		suite.T().Logf("test checkin 1: agent %s", agentID)
+		ackToken, actions := tester.TestCheckin(agentKey, agentID, nil, nil)
+		suite.Require().NotEmpty(actions)
+
+		suite.T().Log("test ack")
+		tester.TestAcks(agentKey, agentID, actions)
+
+		suite.T().Logf("test checkin 2: agent %s 3m timout", agentID)
+		dur := "3m"
+		tester.ctx, cancel = context.WithTimeout(ctx, 3*time.Minute)
+		defer cancel()
+
+		tester.TestCheckin(agentKey, agentID, ackToken, &dur)
+
+		// sanity check agent status in kibana
+		suite.AgentIsOnline(ctx, agentID)
+	})
+
+	suite.Run("test file upload", func() {
+		ctx, cancel := context.WithCancel(bCtx)
+		defer cancel()
+		tester := &ClientAPITester{
+			suite.Suite,
+			ctx,
+			suite.client,
+			"https://localhost:8220",
+		}
+		agentID, agentKey := tester.TestEnroll(enrollmentKey)
+		actionID := suite.RequestDiagnosticsForAgent(ctx, agentID)
+
+		tester.TestFullFileUpload(agentKey, agentID, actionID, 8192) // 8KiB file
+	})
+
+	suite.Run("test artifact", func() {
+		ctx, cancel := context.WithTimeout(bCtx, 3*time.Minute)
+		defer cancel()
+		tester := &ClientAPITester{
+			suite.Suite,
+			ctx,
+			suite.client,
+			"https://localhost:8220",
+		}
+		_, agentKey := tester.TestEnroll(enrollmentKey)
+		suite.AddSecurityContainer(ctx)
+		suite.AddSecurityContainerItem(ctx)
+
+		hits := suite.FleetHasArtifacts(ctx)
+		tester.TestArtifact(agentKey, hits[0].Source.Identifier, hits[0].Source.DecodedSHA256)
```

Comment:
Before these steps are ran, the test setup calls the setup methods for fleet in kibana, and manually inserts the endpoint package